devito/finite_differences/derivative.py

```diff
@@ -171,18 +174,21 @@ def _process_kwargs(cls, expr, *dims, **kwargs):
                           for s in filter_ordered(new_dims)]
         return new_dims, orders, fd_orders, variable_count
 
-    def __call__(self, x0=None, fd_order=None, side=None):
+    def __call__(self, x0=None, coeffs=None, fd_order=None, side=None):
         if self.ndims == 1:
             _fd_order = fd_order or self._fd_order
             _side = side or self._side
             new_x0 = {self.dims[0]: x0} if x0 is not None else self.x0
-            return self._new_from_self(fd_order=_fd_order, side=_side, x0=new_x0)
+            new_coeffs = coeffs
```

Comment:
`coeffs or self._coeffs` ?

Reply:
Need to check - something wasn't working last night so wrote it in this way for now but quite a few things in there need checking and fixing still.